src/tarantool/bank.clj

```diff
@@ -27,7 +27,10 @@
 
   (setup! [this test node]
     (locking BankClientWithLua
-      (let [conn (cl/open node test)]
+      (let [conn (cl/open node test)
+            ; Distribute initial balances uniformly.
+            initial-balance-per-account (/ (:total-amount test)
+                                           (count (:accounts test)))]
```

Comment:
What will happen when user will pass float `total-amount` or/and `accounts` to test?
What will happen when user will pass an `account` equal to zero?

Reply:
> What will happen when user will pass float `total-amount`

Initially I had idea to distribute balances almost uniformly, when it is not possible to do that in a strictly uniform way. Say, for total-balance 10 and three accounts assign balances to `[4 4 2]`.

```lua
local left_amount = total_amount
local usual_balance = math.ceil(total_amount / #accounts)
local balances = {}
for i = 1, #accounts - 1 do
    balances[i] = usual_balance
    left_amount = left_amount - usual_balance
end
balances[#accounts] = left_amount
```

But, to be honest, I don't know Clojure good enough to don't make such code look like mess.

I thought about consequences of just 'balance is total_amount/#accounts` and, I guess, everything should work well: we'll just fill accounts with floating point numbers. Well, I didn't test it, but I doubt that we'll need such parameters in a foreseeable future.

> What will happen when user will pass float <...> `accounts` to test?

Accounts are set as an array, so amount of items cannot be float.

> What will happen when user will pass an `account` equal to zero?

If you meant, an empty array of accounts...

https://github.com/tarantool/tarantool/issues/3077#issuecomment-369734110

Seriously, I'm not very interested about how this code will behave for senseless testing parameters.

Reply:
We discussed it with Sergey and agreed that it worth to add assertion that we'll not fall into floating point numbers.

Reply:
Added assertions.

Reply:
Output in test `bank` with default 'total-amount' and 'accounts' values:
```
INFO [2021-11-24 10:25:53,457] jepsen worker 0 - jepsen.core Running worker 0
INFO [2021-11-24 10:25:53,457] jepsen nemesis - jepsen.core Running nemesis
INFO [2021-11-24 10:25:53,475] jepsen worker 0 - jepsen.util 0  :invoke :read   nil
INFO [2021-11-24 10:25:53,589] jepsen worker 0 - jepsen.util 0  :ok     :read   {0 10, 1 10, 2 10, 3 10, 4 10, 5 10, 6 10, 7 10, 8 10, 9 10}
INFO [2021-11-24 10:25:53,589] jepsen worker 0 - jepsen.util 0  :invoke :read   nil
INFO [2021-11-24 10:25:53,665] jepsen worker 0 - jepsen.util 0  :ok     :read   {0 10, 1 10, 2 10, 3 10, 4 10, 5 10, 6 10, 7 10, 8 10, 9 10}
INFO [2021-11-24 10:25:53,665] jepsen worker 0 - jepsen.util 0  :invoke :read   nil
INFO [2021-11-24 10:25:53,753] jepsen worker 0 - jepsen.util 0  :ok     :read   {0 10, 1 10, 2 10, 3 10, 4 10, 5 10, 6 10, 7 10, 8 10, 9 10}
```

With ':total-amount' equal to 101 and default 'accounts' value test failed:
```
INFO [2021-11-24 10:28:05,897] jepsen node 37.139.42.230 - jepsen.core downloading /var/log/tarantool/jepsen.log to jepsen.log
INFO [2021-11-24 10:28:05,939] jepsen node 37.139.42.230 - tarantool.db 37.139.42.230 Stopping Tarantool
WARN [2021-11-24 10:28:06,693] main - jepsen.core Test crashed!                                                                                               
java.lang.AssertionError: Assert failed: Unable to distribute initial balances uniformly for given total-amount and accounts
(integer? initial-balance-per-account)      
```

Reply:
Thanks for verifying it.